pages/api/fetchData.js

```diff
@@ -1,22 +1,44 @@
-export default async function fetchData() {
-  try {
-    const host = "http://api.wisey.app";
-    const version = "api/v1";
-    const accessToken = await fetch(
-      `${host}/${version}/auth/anonymous?platform=subscriptions`
-    );
-    const accesData = await accessToken.json();
-    const token = accesData.token;
-    const res = await fetch(`${host}/${version}/core/preview-courses`, {
-      headers: {
-        Authorization: `Bearer ${token}`,
-      },
-    });
-    const data = await res.json();
+import fetchAccessToken from "./accessToken";
+class ApiClient {
+  static instance;
+  accessToken;
 
-    return { data: data || null, accesData: accesData || null };
-  } catch (error) {
-    console.error("Error fetching data:", error);
-    throw error;
+  static async getInstance() {
+    if (!ApiClient.instance) {
+      ApiClient.instance = new ApiClient();
+      await ApiClient.instance.refreshToken();
+    }
+    return ApiClient.instance;
+  }
+
+  async refreshToken() {
+    this.accessToken = await fetchAccessToken();
+    // Schedule the next token refresh in 1 hour
+    setTimeout(() => {
+      this.refreshToken();
+    }, 3600 * 1000);
```

Comment:
You can probably use `setInterval` in the `getInstance` method instead. This way you don't have to set the next timeout every time

Reply:
I`ll try to fix it in the next homework!)